cdap-common/src/main/java/io/cdap/cdap/common/http/CommonNettyHttpServiceBuilder.java

```diff
@@ -54,8 +56,21 @@ public void modify(ChannelPipeline pipeline) {
     this.setExceptionHandler(new HttpExceptionHandler());
   }
 
+  /**
+   * Sets pipeline modifier, preserving the security one installed in constructor.
```

Comment:
Isn't this what the addChannelPipelineModifier does??

Reply:
Nope. This call preserves only the one installed in constructor. If called multiple times, previous modifier will be replaced by next one. `addChannelPipelineModifier` collects modifiers, so multiple calls to add will add multiple modifiers

Reply:
To clarify: I changed this call for two reasons:
1) It's much easier to use it in the chained calls as it's an override
2) From what I saw it was often used wrongly without full understanding that by this call you remove security part of pipeline. The way it is now is more natural - you don't care how security is injected into pipeline, with some kind of modifier or not (it's implementation detail), you can simply set your own modifier preserving all `CommonNettyHttpServiceBuilder` behavior